source/fundamentals/data-formats/document-data-format-record.txt

```diff
@@ -0,0 +1,141 @@
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to store and retrieve data in the {+driver-long+}
+using **Java records**. Java records are a type of Java class often used to
+model data and separate business logic from data representation.
+
+.. tip::
+
+   You can declare Java records in Java 14 or later. Learn more about the
+   functionality and restrictions of records from `Java 14 Language Updates: Records <https://docs.oracle.com/en/java/javase/14/language/records.html>`__.
+
+   If you are using an earlier version of Java, you can use plain old Java
+   objects instead. See the :ref:`<fundamentals-pojos>` guide for
+   implementation details.
+
+The following sections show how you can perform the following tasks:
+
+- Configure the driver to serialize and deserialize a record.
+- Read data from MongoDB into records.
+- Write data from records into MongoDB.
+
+.. _fundamentals-example-record:
+
+Example Record
+--------------
+
+The code examples in this guide reference the following sample record, which
+describes a data storage device:
+
+.. literalinclude:: /includes/fundamentals/code-snippets/records/DataStorageRecord.java
+   :language: java
+   :start-after: start dataStorageRecord
+   :end-before: end dataStorageRecord
+
+Serialize and Deserialize a Record
+----------------------------------
+
+To set up the driver to serialize and deserialize a record, you must provide
+it with logic on how to convert between Java and MongoDB types using
+**codecs**. Codecs define how to encode and decode data between records and
+MongoDB documents, and to which classes or packages the codecs they apply.
+The driver defines methods for the encoding and decoding logic in an interface
+called ``Codec``.
+
+.. tip::
+
+   Learn more about how to create and manage codecs from the guide on
+   :ref:`<fundamentals-codecs>`.
+
+To identify which codec to use, the driver uses a registry called the
+``CodecRegistry``. The ``CodecRegistry`` is an immutable collection of
+``Codec`` implementations that identify which object type they can convert.
+
+To specify the scope in which to apply the codecs in the registry, call
+the ``withCodecRegistry`` method on an instance of either a ``MongoDatabase``
+or ``MongoCollection``.
+
+The driver provides the ``DefaultCodecRegistry``, a codec registry that
+includes ``CodecRegistry`` classes that contain commonly-used ``Codec``
+implementations.
+
+See the following code for an example of how to set the
+``DefaultCodecRegistry`` on the ``MongoDatabase`` instance:
+
+.. code-block:: java
+   :emphasize-lines: 7,8
+
+   import static com.mongodb.MongoClientSettings.getDefaultCodecRegistry;
+   import static org.bson.codecs.configuration.CodecRegistries.fromRegistries;
+
+   // ...
+
+   try (MongoClient mongoClient = MongoClients.create(uri)) {
+       CodecRegistry codecRegistry = fromRegistries(getDefaultCodecRegistry());
```

Comment:
This is a no-op, since the driver will use the default codec registry by default (hence the name! ;) ).  I think it can be removed.